hs_core/models.py

```diff
@@ -1447,55 +1454,46 @@ def scimeta_url(cls, resource_id):
         scimeta_url = reverse('rest_download', kwargs={'path': scimeta_path})
         return scimeta_url
 
+    # TODO: there are too many ways to get to the resourcemap.
+    # 1. {id}/data/resourcemap.xml
+    # 2. {id}/resmap
+    # Choose one!
     @classmethod
     def resmap_url(cls, resource_id):
         resmap_path = "{resource_id}/data/resourcemap.xml".format(resource_id=resource_id)
         resmap_url = reverse('rest_download', kwargs={'path': resmap_path})
         return resmap_url
 
+    # TODO: this is inaccurate; resourcemap.xml != systemmetadata.xml
     @classmethod
     def sysmeta_path(cls, resource_id):
         return "{resource_id}/data/resourcemap.xml".format(resource_id=resource_id)
 
     def delete(self, using=None):
         from hydroshare import hs_bagit
         for fl in self.files.all():
-            if fl.fed_resource_file_name_or_path:
-                istorage = IrodsStorage('federated')
-                # ensure fed_res_file_path holds relative path in the format of data/contents/...
-                # so that a uniform path can be formed by concatenating with self.root_path for
-                # file deletion by irods storage
-                fed_res_file_path = fl.fed_resource_file_name_or_path
-                idx = fed_res_file_path.find(self.short_id)
-                if idx >= 0:
-                    s_idx = idx + len(self.short_id) + 1
-                    fed_res_file_path = fed_res_file_path[s_idx:]
-                istorage.delete(os.path.join(self.root_path, fed_res_file_path))
-            elif fl.resource_file:
-                fl.resource_file.delete()
-            elif fl.fed_resource_file:
-                fl.fed_resource_file.delete()
-
             if fl.logical_file is not None:
                 # delete of metadata file deletes the logical file (one-to-one relation)
                 # so no need for fl.logical_file.delete() and deleting of metadata file
                 # object deletes (cascade delete) all the contained GenericRelated metadata
                 # elements
                 fl.logical_file.metadata.delete()
-
+            # COUCH: delete of file objects now cascades.
             fl.delete()
```

Comment:
@alvacouch Some files such as resourcemap.xml, resourcemetadata.xml, netCDF header file, raster vrt file, etc., are not stored in ResourceFile class, but present in iRODS, so ResourceFile delete() alone is not enough to delete all files for a resource. delete_bag() routine (I believe you renamed it in this PR) which is called by resource deletion action actually deletes the whole resource collection on iRODS side so that all these files that are not represented in ResourceFile class are deleted as a result of resource deletion. So the current implementation does not have this file leak (if I understand it correctly) issue and may be sufficient for our needs.